PSReadLine/UndoRedo.cs

```diff
@@ -278,7 +278,7 @@ public static EditItem Create(string str, int position, Action<ConsoleKeyInfo?,
             public override void Undo()
             {
                 _singleton._buffer.Insert(_deleteStartPosition, _deletedString);
-                _singleton._current = _deleteStartPosition + _deletedString.Length;
+                _singleton._current = _deleteStartPosition;
```

Comment:
This change doesn't seem right for some basic text-deleting scenario, for example, in "Windows" edit mode:

_Before the change_, selecting part of text from left to right, then "Delete", then "Ctrl+z"
![demo1](https://user-images.githubusercontent.com/127450/103954174-8184af00-50f8-11eb-876d-c51fd1d3b0cb.gif)

Or, selecting from right to left, then "Delete", then "Ctrl+z"
![demo1](https://user-images.githubusercontent.com/127450/103954111-5f8b2c80-50f8-11eb-9fd1-b6666959b793.gif)

_After the change_, the cursor will always at <kbd>c</kbd> after "Ctrl+z"

This is a pretty noticeable breaking change. Personally, I'm very used to having "Ctrl+z" to place the curosr at the end for scenarios like the above 2 examples, especially the 2nd one.

------------

Maybe, what we need is something like `_insertAnchor` in `EditItemInsertLines`, to instruct where `EditItemDelete.Undo` should place the cursort at. Maybe just a `bool` field to indicate whether the cursor should be placed at the beginning position of the deleted text, or the end of it (default should be the end so no breaking changes).

Reply:
I’ve just added a `bool adjustCursor` flag as you suggested to the `EditItemDelete.Create` method.
This flag defaults to `true` meaning that the cursor moves at the end of the re-inserted text after undoing a delete operation.

In VI-mode specific files, such as `ReadLine.vi.cs` and `Replace.vi.cs`, this flag is explicitely overridden and set to `false`.

In other parts of the code, where the operation is shared between Vi-mode and Windows-mode, this flag is computed using the expression `Options.EditMode != EditMode.Vi` which keeps the behaviour unchanged for Windows (an potentially Emacs) mode.

Finally, a call in `Kill` has not been changed, thus taking advantage of the default value to keep the existing behaviour unchanged for Emacs-mode.